internal/mode/static/handler.go

```diff
@@ -294,64 +294,89 @@ func (h *eventHandlerImpl) waitForStatusUpdates(ctx context.Context) {
 			return
 		}
 
-		// TODO(sberman): once we support multiple Gateways, we'll have to get
-		// the correct Graph for the Deployment contained in the update message
+		// get the deploymentName from the item
+		deploymentName := item.Deployment
+		if deploymentName == (types.NamespacedName{}) {
```

Comment:
If deployment name is empty, we still need to update gatewayclass status. Look to see where we call Enqueue, you'll notice a few that don't set the name.